lib/msal-core/test/ScopeSet.spec.ts

```diff
@@ -1,16 +1,205 @@
-import { expect } from "chai";
-import sinon from "sinon";
 import { ScopeSet } from "../src/ScopeSet";
+import { expect } from "chai";
+import { ClientConfigurationError, ClientConfigurationErrorMessage } from "../src/error/ClientConfigurationError";
+import { Constants } from "../src/utils/Constants";
+import { TEST_CONFIG } from "./TestConstants";
+
+describe("ScopeSet.ts", () => {
+    const openid = Constants.openidScope;
+    const profile = Constants.profileScope;
+    const clientId = TEST_CONFIG.MSAL_CLIENT_ID;
+
+    describe("isIntersectingScopes", () => {
+        it("should return true if cachedScopes and scopes share a single scope", () => {
+            const cachedScopes = ["s1"];
+            const scopes = ["s1"];
+            expect(ScopeSet.isIntersectingScopes(cachedScopes, scopes)).to.eql(true);
+        });
+
+        it("should return true if cachedScopes and scopes share a single scope while having other scopes", () => {
+            const cachedScopes = ["s1", "s2", "s3"];
+            const scopes = ["s4", "s1", "s5"];
+            expect(ScopeSet.isIntersectingScopes(cachedScopes, scopes)).to.eql(true);
+        });
+
+        it("should return true if cachedScopes and scopes share more than one scope", () => {
+            const cachedScopes = ["s1", "s2", "s3"];
+            const scopes = ["s2", "s1", "s5"];
+            expect(ScopeSet.isIntersectingScopes(cachedScopes, scopes)).to.eql(true);
+        });
+
+        it("should return false if cachedScopes and scopes do not share any scopes", () => {
+            const cachedScopes = ["s1"];
+            const scopes = ["s2"];
+            expect(ScopeSet.isIntersectingScopes(cachedScopes, scopes)).to.eql(false);
+        });
+
+        it("should return false if scopes is empty", () => {
+            const cachedScopes = ["s1"];
+            const scopes = [];
+            expect(ScopeSet.isIntersectingScopes(cachedScopes, scopes)).to.eql(false);
+        });
+
+        it("should return false if cachedScopes and scopes are both empty arrays", () => {
+            const cachedScopes = [];
+            const scopes = [];
+            expect(ScopeSet.isIntersectingScopes(cachedScopes, scopes)).to.eql(false);
+        });
+    });
+
+    describe("trimAndConvertToLowerCase", function() {
+        it("should downcase and remove all whitespace from all scope strings in scopes array passed in", () => {
+            const scopeSet = ["S1", " S2", " S3 "];
+            expect(ScopeSet.trimAndConvertArrayToLowerCase(scopeSet)).to.be.deep.eq(["s1", "s2", "s3"]);         
+        });
+    });
+
+    describe("trimAndConvertToLowerCase", () => {
+        it("should downcase and remove all whitespace from the scope passed in", () => {
+            const scope = " S1 ";
+            expect(ScopeSet.trimAndConvertToLowerCase(scope)).to.be.eq("s1");
+        });
+    });
+
+    describe("removeElement", function() {
```

Comment:
Add a test for output array = input array if `scopeToRemove` is not in the input array

Reply:
Done.